### ISSUE TITLE
Other potential sources of test corpora

### ISSUE THREAD

**Issue by gvanrossum** [we-like-parsers/pegen#246]
I [tweeted](https://twitter.com/gvanrossum/status/1297687557715529728) about this and got a few tips that I might have to follow up if the current corpus is not sufficiently useful.

- [ ] [Meg Risdal from Kaggle might have some](https://twitter.com/MeganRisdal/status/1297692573260083200)
- [ ] I got a [tip](https://twitter.com/YadKonrad/status/1297698662055735297) about [Buggify](https://github.com/roovyshapiro/Buggify) (looks immature)
- [ ] [Apparently](https://twitter.com/dmoisset/status/1297819956676108288) @Tobias-Kohn may have [something](https://pretalx.com/pyconuk-2019/talk/G93RFU/)

**Comment by Tobias-Kohn** [comment]
Yes, I do have several thousand programs.  What exactly are you looking for: just the programs, or labelled data?

If a summary is helpful: I reported my findings concerning errors in a paper [The Error Behind the Message](https://dl.acm.org/doi/10.1145/3287324.3287381), which is also available from my [homepage](https://tobiaskohn.ch/index.php/research/publications/).

Just as a disclaimer: the programs I have collected are all anonymised (before they were sent to me), i.e. the strings and comments basically comprise only `x`s, e.g., `s = "Xxxx x xx xxxx", and large numbers are scrambled.  Since I am interested in syntax errors and conceptual misunderstandings, this works just fine for me.  However, some of the runtime errors might be quite distorted.

We could also ask Philip Guo, who might have some data from his [Python Tutor](http://www.pythontutor.com/).

**Comment by Tobias-Kohn** [comment]
By the way: my [Python-parser](https://github.com/Tobias-Kohn/TigerPython-Parser) comes with a very small test suite, which contains some [erroneous programs](https://github.com/Tobias-Kohn/TigerPython-Parser/tree/master/tpParser/shared/src/test/programs/erroneous).  They are in a format where the first two lines contain the expected number of the erroneous line and the expected error message, followed by a small program.  Would that kind of format work for you?

**Comment by gvanrossum** [comment]
Yes, that format works for me. It will encourage me to stop working with straight JSON and just load test cases into a sqlite3 database. :-) I notice that your "expected error message" field is some kind of enum. Do you have a way to map these to what CPython produces? (No worries if you don't.)
